### PR TITLE
Coriolis linear solver

### PR DESCRIPTION
**Pull request by Witt-D**
Adding the Coriolis term to be inside the linear solvers

### CODE REVIEW

File: gusto/solvers/linear_solvers.py [hunk 1]
```diff
@@ -316,14 +316,17 @@ def L_tr(f):
...
-            + dl*dot(u, n)*(ds_tbp + ds_vp)
-        )
+            + dl*dot(u, n)*(ds_tbp + ds_vp))
 
         # TODO: can we get this term using FML?
         # contribution of the sponge term
         if hasattr(self.equations, "mu"):
             eqn += dt*self.equations.mu*inner(w, k)*inner(u, k)*dx
 
+        if equations.parameters.coriolis is not None:
```
Comment (tommbendall) [inline gusto/solvers/linear_solvers.py:326]:
```suggestion
        if equations.parameters.Omega is not None:
```

File: gusto/solvers/linear_solvers.py [hunk 2]
```diff
@@ -652,8 +660,11 @@ def _setup_solver(self):
             - beta_u * 0.5 * bbar * div(w*(D-Dbar)) * dx
             + beta_u * 0.5 * jump((D-Dbar)*w, n) * avg(bbar) * dS
             + inner(phi, (D - D_in)) * dx
-            + beta_d * phi * Dbar * div(u) * dx
-        )
+            + beta_d * phi * Dbar * div(u) * dx)
+
+        if self.prescribed_fields('coriolis'):
```
Comment (tommbendall) [inline gusto/solvers/linear_solvers.py:665]:
```suggestion
        if 'coriolis' in self.prescribed_fields._field_names:
```

File: gusto/solvers/linear_solvers.py [hunk 3]
```diff
@@ -867,6 +878,10 @@ def _setup_solver(self):
             + beta_d * phi * Dbar * div(u) * dx
         )
 
+        if self.prescribed_fields('coriolis'):
```
Comment (tommbendall) [inline gusto/solvers/linear_solvers.py:881]:
```suggestion
        if 'coriolis' in self.prescribed_fields._field_names:
```

**Review by tommbendall — CHANGES_REQUESTED** [review]

File: gusto/solvers/linear_solvers.py [hunk 4]
```diff
@@ -316,14 +316,17 @@ def L_tr(f):
             # through the interior facets and weakly impose the no-slip
             # condition
             + dl('+')*jump(u, n=n)*(dS_vp + dS_hp)
-            + dl*dot(u, n)*(ds_tbp + ds_vp)
-        )
+            + dl*dot(u, n)*(ds_tbp + ds_vp))
```
Comment (tommbendall) [inline gusto/solvers/linear_solvers.py:319]: Could you revert this bracket change?

File: gusto/solvers/linear_solvers.py [hunk 5]
```diff
@@ -652,8 +660,11 @@ def _setup_solver(self):
             - beta_u * 0.5 * bbar * div(w*(D-Dbar)) * dx
             + beta_u * 0.5 * jump((D-Dbar)*w, n) * avg(bbar) * dS
             + inner(phi, (D - D_in)) * dx
-            + beta_d * phi * Dbar * div(u) * dx
-        )
+            + beta_d * phi * Dbar * div(u) * dx)
```
Comment (tommbendall) [inline gusto/solvers/linear_solvers.py:663]: and also this bracket change?

**Review by jshipton — APPROVED** [review]
great - thanks @Witt-D !

**Pull request merged**